POC: Patching v2 (#20)
ai-platform-infrastructure/open-webui-docker: README.md

## Patching

> This project requires us to make changes requested by the upstream project to align with the organization. The upstream

Reviewer: "This project requires us to make changes requested by the _upstream project_ to align with the organization. The _upstream project_ may not appreciate some of these changes"

So "upstream" don't want their own changes? Not sure what the intended message is but this should propably be re-written.

project may not appreciate some of these changes, so we need to be able to keep this project in sync with upstream and
at the same time track local changes/patches to the codebase.

> We use branches and "local" pull request to track changes and go-task to manage the patches. The pull requests can be

Reviewer: ```suggestion
We use branches and "local" pull request against our own fork to track changes and go-task to manage the patches. The pull requests can be
```

found at [https://github.com/itk-dev/open-webui/pulls](https://github.com/itk-dev/open-webui/pulls) notice the use of
labels to mark PR's as patches when approved and to mark which are pending upstream.

### Update pull requests

> When a new release is, well released, upstream, we need to update the patches to the new release.

Reviewer: ```suggestion
When a new upstream release is published, we need to update the patches to the new release.
```

But first we need to update dev and main branches with upstream. This is easily done on GitHub. But as GitHub does not
synchronize the tags, the following task can be used to do that.

```shell
task git:sync:tags
```

> Then we can update the patches. This requires some manuale steps to ensure that the patches can be applied cleanly.

Reviewer: ```suggestion
Then we can update the patches. This requires some manual steps to ensure that the patches can be applied cleanly.
```
